homestar-wasm/README.md

```diff
@@ -23,15 +23,1054 @@
+  variant where none of the cases have a payload type.
+
+  * **IPLD to WIT Translation**
+
+    When a WIT function expects an `enum` input, an `Ipld::String` value is
+    mapped to a `enum` WIT runtime value.
+
+    **Example**:
+
+    ```wit
+    enum color {
+        Red,
+        Green,
+        Blue
+    }
+
+    export fn: func(a: color) -> string;
+    ```
+
+    Given an JSON input for this function:
+
+    ```json
+    {
+      "args": ["Green"]
+    }
+    ```
+
+    `"Green"` is converted into an `Ipld::String`, which is then translated and
+    passed into `fn` as a enum argument (`color`). **You'll have to provide a
+    singular string that matches on one of the discriminants**.
+
+  * **WIT to IPLD Translation**:
+
+    Conversely, when an enum value is returned from a WIT function, it can be
+    translated back into an `Ipld::String` value.
+
+**IPLD Schema Definitions**:
+
+``` ipldsch
+type Enum enum {
+  | Red
+  | Green
+  | Blue
+}
+
+type IPLDStringAsWit union {
+ | Enum     Enum
+ | String   string
+ | Char     string
+ | Listu8In string
+} representation kinded
+
+type WitAsIpldString union {
+ | Enum      Enum
+ | String    string
+ | Char      string
+ | Listu8Out bytes
+} representation kinded
+```
+
+#### Bytes
+
+This section outlines the translation process between IPLD bytes values
+(`Ipld::Bytes`) and varying [WIT runtime values][wit-val]. A `Ipld::Bytes` value
+can be interpreted either as a `list<u8>` or `string`.
+
+- [`list`][wit-list]:
+
+  * **IPLD to WIT Translation**
+
+    When a WIT function expects a `list<u8>` input, an `Ipld::Bytes` value is
+    mapped to a `list<u8>` WIT runtime value.
+
+    **Example**:
+
+    ```wit
+    export fn: func(a: list<u8>) -> list<u8>;
+    ```
+
+    Given an JSON input for this function:
+
+    ```json
+    {
+      "args": [{"/": {"bytes": "aGVsbDA"}}]
+    }
+    ```
+
+    `"aGVsbDA"` is converted into an `Ipld::Bytes`, which is then translated
+    into bytes and passed into `fn` as a `list<u8>` argument.
+
+  * **WIT to IPLD Translation**:
+
+    Conversely, when a `list<u8>` value is returned from a WIT function, it is
+    translated back into an `Ipld::Bytes` value if the list contains valid
+    `u8` values.
+
+- `string`
+
+  * **IPLD to WIT Translation**
+
+    When a WIT function expects a `string` input, an `Ipld::Bytes` value is
+    mapped to a `string` WIT runtime value.
+
+    **Example**:
+
+    ```wit
+    export fn: func(a: string) -> string;
+    ```
+
+    Given an JSON input for this function:
+
+    ```json
+    {
+      "args": [{"/": {"bytes": "aGVsbDA"}}]
+    }
+    ```
+
+    `"aGVsbDA"` is converted into an `Ipld::Bytes`, which is then translated
+    into a `string` and passed into `fn` as a `string` argument.
+
+  * **WIT to IPLD Translation**:
+
+    **Here, when a string value is returned from a WIT function, it is
+    translated into an `Ipld::String` value, as we can't determine if
+    it's related to `bytes` once again**.
```

Comment:
```suggestion
    **Here, when a string value is returned from a WIT function, it is
    translated into an `Ipld::String` value, because we can't determine if it
    was originally `bytes`**.
```
Possible rephrasing.